### ISSUE TITLE
--omit="./*.py" omits ALL python files, not only files at the top level

### ISSUE THREAD

**Issue by Kache** [coveragepy/coveragepy#1407]
**Describe the bug**
My intent is to trace all python files under project root, excepting a couple directories and python files in the top level.

This is close to what I want, but contains python files at the top level:
```sh
coverage run --source=. --omit="tests/*","tests_integ/*" -m pytest
```

but these will omit ALL python files (`CoverageWarning: No data was collected. (no-data-collected)`):
```
coverage run --source=. --omit="tests/*,tests_integ/*,*.py" -m pytest
coverage run --source=. --omit="tests/*,tests_integ/*,./*.py" -m pytest
coverage run --source=. --omit='tests/*,tests_integ/*,*.py' -m pytest
```

**To Reproduce**
How can we reproduce the problem?  Please *be specific*. Don't link to a failing CI job.

Minimal repro:

```
❯ tree -I "__*"
.
├── bar
│   └── bar.py
├── foo.py
└── tests
    └── test_baz.py

2 directories, 3 files
```

```
❯ bat foo.py bar/bar.py tests/test_baz.py
───────┬────────────────────────────────────
       │ File: foo.py   <EMPTY>
───────┴────────────────────────────────────
───────┬────────────────────────────────────
       │ File: bar/bar.py   <EMPTY>
───────┴────────────────────────────────────
───────┬────────────────────────────────────
       │ File: tests/test_baz.py
───────┼────────────────────────────────────
   1   │ def test_foo():
   2   │     assert True
───────┴────────────────────────────────────
```

Then compare `coverage report` after running the two different commands:

```sh
coverage run --source=. -m pytest  # traces coverage for every file
coverage run --source=. --omit="./*.py" -m pytest  # no files traced
```

Answer the questions below:
1. What version of Python are you using? `3.7.6`
1. What version of coverage.py shows the problem? The output of `coverage debug sys` is helpful. `6.4.1`
1. What versions of what packages do you have installed? The output of `pip freeze` is helpful.
1. What code shows the problem?  Give us a specific commit of a specific repo that we can check out. If you've already worked around the problem, please provide a commit before that fix.
1. What commands did you run?

**Expected behavior**
I expected either `*.py` or `./*.py` to only match python files at the top level. In minimal repro above, that means `./foo.py` should have been omitted from coverage trace, but `./bar/bar.py` should not.

**Additional context**
I've also tried "omitting on read" via `coverage report --omit`, but that has the same problem.

**Comment by AvinashUtekar** [comment]
command :   **coverage run --source = . -m pytest**
error : Can't find '__main__' module in '.'

can you please help me, why i am getting this error?

**Comment by nedbat** [comment]
@Kache I see what you mean.  I'm using fnmatch.translate() to turn file name patterns into regexes, and it converts `"./*.py"` into `r"\./.*\.py"`, and the `".*"` in the middle matches directory separators.  This is because fnmatch is only meant for matching file names and not full file paths. I'm using it incorrectly, and I need to do something about that.

**Comment by nedbat** [comment]
This is now fixed in commit ec6205a8.

**Comment by nedbat** [comment]
This is now released as part of [coverage 6.6.0b1](https://pypi.org/project/coverage/6.6.0b1).

**Comment by nedbat** [comment]
This is now released as part of [coverage 7.0.0b1](https://pypi.org/project/coverage/7.0.0b1).

**Comment by webknjaz** [comment]
FTR with this change, a trailing `/*` is treated as `/**` at the very end of path patterns. Not sure if this was intended, but it's not documented and isn't tested, it seems. Trying to figure it out...

https://github.com/nedbat/coveragepy/commit/ec6205a8de972af6a09453235d02a7ebea6aea8e#r121300002.

**Comment by webknjaz** [comment]
It looks like it's forbidden to have a wildcard at the end of `paths` https://github.com/nedbat/coveragepy/blob/7.2.7/tests/test_files.py#L483-L492, but this should probably work for `omit`/`include`, it seems.

**Comment by webknjaz** [comment]
I submitted https://github.com/nedbat/coveragepy/pull/1650 in hopes of clearing up the confusion...

**Comment by nedbat** [comment]
This is now released as part of [coverage 7.3.1](https://pypi.org/project/coverage/7.3.1).
